src/components/BookSections/BookSections.test.tsx

```diff
@@ -0,0 +1,11 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import App from "../../App";
+
+describe("Test book number in section", () => {
+  test("should show number of books returned from server when number is provided", async () => {
+    render(<App />);
+    await waitFor(() => {
```

Comment:
Use `find*` any time you want to query for something that may not be available right away, in your case it would be something like this: 
`expect(await screen.findByText('Catalog (15)')).toBeTruthy();`

`find*` queries use `waitFor` under the hood, they are simpler and the error message you get will be better.

Please check here for more info https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find